modules/transactionpool/update.go

```diff
@@ -60,60 +171,69 @@ func (tp *TransactionPool) ProcessConsensusChange(cc modules.ConsensusChange) {
 			}
 		}
 
-		// Add the transactions from this block.
-		var totalSize uint64
-		for _, txn := range block.Transactions {
+		// Find the median transaction fee for this block.
+		type feeSummary struct {
+			fee  types.Currency
+			size int
+		}
+		var fees []feeSummary
+		var totalSize int
+		txnSets := findSets(block.Transactions)
+		for _, set := range txnSets {
+			// Compile the fees for this set.
 			var feeSum types.Currency
-			size := uint64(len(encoding.Marshal(txn)))
-			for _, fee := range txn.MinerFees {
-				feeSum = feeSum.Add(fee)
+			var sizeSum int
+			for _, txn := range set {
+				sizeSum += len(encoding.Marshal(txn))
```

Comment:
I doubt the performance impact would be very large. If we need maximum performance we'd want to write a custom `(Transaction).MarshalSiaLen` method that calculates the length without allocating or copying any memory.